src/components/Map/Map.js

```diff
@@ -0,0 +1,23 @@
+import './Map.css';
+import { useEffect } from 'react';
+import { mapService } from '../../services/mapService';
+
+const Map = () => {
+  useEffect(() => {
+    window.addEventListener('load', () => {
+      mapService.initMap();
+    });
+
+    return () => {
+      window.removeEventListener('load', () => {
+        mapService.initMap();
+      });
+    }
+  });
+
+  return (
+    <canvas id="map"></canvas>
+  )
```

Comment:
So, `useEffect` runs after the render. This ensures that the `mapService.initMap()` function is called after there is a canvas with the required id in there. But I wonder if you cannot remove the `window.addEventListener` and `window.removeEventListener` parts and just call `initMap` directly in the `useEffect` callback.